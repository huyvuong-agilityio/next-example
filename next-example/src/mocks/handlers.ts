import { rest } from "msw";

export const handlers = [
  rest.get("/api/user", (req, res, ctx) => {
    return res(ctx.status(200), ctx.json({ name: "Vuong" }), ctx.delay(1000));
  }),
];
