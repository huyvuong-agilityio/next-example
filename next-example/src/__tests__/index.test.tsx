import { render, screen, waitFor } from '@testing-library/react';
import Home from '@pages/index';
import { userEvent } from '@storybook/testing-library';
import { handlers } from '@mocks/handlers';
import { server } from '@mocks/server';

describe('Home', () => {
  test('renders homepage unchanged', () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  test('renders a heading', async () => {
    server.use(...handlers);

    render(<Home />);

    const button = screen.findByRole('button');
    button.then((element) => {
      userEvent.click(element);
    });

    await waitFor(
      () =>
        screen.findByRole('heading', {
          name: /vuong/i,
        }),
      {
        timeout: 2000,
      },
    );

    expect(
      screen.getByRole('heading', {
        name: /vuong/i,
      }),
    ).toBeInTheDocument();
  });
});
