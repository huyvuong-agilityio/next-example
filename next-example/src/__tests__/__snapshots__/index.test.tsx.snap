// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <div>
    <h1>
      name
    </h1>
    <button>
      Get user
    </button>
  </div>
</div>
`;
